src/api/app/models/project.rb

```diff
@@ -1245,14 +1246,34 @@ def update_product_autopackages
   end
 
   def open_requests
-    reviews = BsRequest.collection(project: name, states: %w(review)).map(&:number)
-    targets = BsRequest.collection(project: name, states: %w(new)).map(&:number)
-    incidents = BsRequest.collection(project: name, states: %w(new), types: %w(maintenance_incident)).map(&:number)
-
+    reviews = BsRequest.with_involved_projects(id)
+                       .or(BsRequest.from_source_project(name))
+                       .with_actions_and_reviews
+                       .where(state: :review, reviews: { state: :new })
+                       .pluck(:number)
+
+    targets = BsRequest.with_involved_projects(id)
+                       .or(BsRequest.from_source_project(name))
+                       .in_states(:new).with_actions
+                       .pluck(:number)
+
+    incidents = BsRequest.with_involved_projects(id)
+                         .or(BsRequest.from_source_project(name))
+                         .in_states(:new)
+                         .with_types(:maintenance_incident)
```

Comment:
Hm, we could already put the joins in the ``with_types`` and you can remove the ``with_actions`` then because the ``with_types`` is anyway invalid without the joins, right?